code/services/QueuedJobService.php

```diff
@@ -206,15 +206,10 @@ protected function copyDescriptorToJob($jobDescriptor, $job) {
 		$jobData = @unserialize($jobDescriptor->SavedJobData);
 		$messages = @unserialize($jobDescriptor->SavedJobMessages);
 
-		if (!$jobData) {
-			// SS's convert:: function doesn't do this detection for us!!
-			if (function_exists('json_decode')) {
-				$jobData = json_decode($jobDescriptor->SavedJobData);
-				$messages = json_decode($jobDescriptor->SavedJobMessages);
-			} else {
-				$jobData = Convert::json2obj($jobDescriptor->SavedJobData);
-				$messages = Convert::json2obj($jobDescriptor->SavedJobMessages);
-			}
+		// try decoding as json if null
+		if (function_exists('json_decode')) {
+			$jobData = $jobData ?: json_decode($jobData);
```

Comment:
Does this need to be decoding `$jobDescriptor->SavedJobData`?  Otherwise if `$jobData` is false from a failure in `unserialize()` then `json_decode(false)` won't help us

Reply:
*cough* I don't know what you mean >_>